### ISSUE TITLE
Issue: Error: file should be valid json

### ISSUE THREAD

**Issue by michahell** [sverweij/dependency-cruiser#667]
In a (MWE) Angular project with:

- ESLint (using `.eslintrc.cjs` config file)
- lint-staged
- ng-lint-staged (`ng lint` file path transformer for `lint-staged`)
- husky
- depcruiser

When running `depcruiser` as part of a bunch of checks that need to be done only for staged files,
while using the `--ignore-known` flag and having generated a baseline with a bunch of legacy errors,
with a single angular `.ts` file changed,
depcruiser seems to be quitting with an error indicating the scanned `.ts` file is not valid JSON.

At first I thought this had something to do with lint-staged and the way that filepaths are presented to lint-staged tasks, but this is not the case, as it has worked before, with a slimmed down depcruiser config.

I have tried slimming my own (actual project, not MWE) depcruiser down, but I can't seem to figure out what causes this.
while I am filing this bug report, I'll continue trying to hunt down the cause, but I thought it would have value to have this here too, in case others run into this particular problem too.

### Expected Behavior

depcruiser should run succesfully and report finding zero issues (the baseline contains all of the issues existing in a greenfield new Angular 14.1 project)

### Current Behavior

The following error is thrown:
<img width="1022" alt="depcruiser-lint-staged" src="https://user-images.githubusercontent.com/225784/191313776-d9cfc209-2c70-40ce-a0bb-2e831db83e24.png">

### Possible Solution

As stated before, it's not lint-staged, this combination works 🪄 ✨ 💖 
- Slimming down the depcruiser config until problem is found might work
- Being able to debug depcruiser while scanning will also help

### Steps to Reproduce (for bugs)

See MWE repo here: https://github.com/michahell/depcruiser-json-error

1. check out project
2. change a line, add a console.log or something in an Angular component  `.ts` file
3. git commit
4. husky is triggered, lint-staged runs, depcruiser runs
5. but should fail with the error in the screenshot above

### Context

I am trying to have depcruiser run only on staged files with a default `--init` depcruiser config modified with some custom rules

### Your Environment

- Version used: 11.15
- Node version: v16.15.0 / npm 8.19.2
- Operating System and version: OSX 12.6
- Link to your project: https://github.com/michahell/depcruiser-json-error (MWE)

**Comment by sverweij** [comment]
hi @michahell long time no see, how are you doing :-) 

In any case - thanks for raising this issue and for the clear minimal reproduction sample!

### TL;DR 

- Add ` --` at the end of the depcruise line in the `.lintstagedrc`
- You might want to tweak a few other things as well (keep on reading ...)

### What's happening?

`lint-staged` executes each command in the array and pastes matching files behind it. For the depcruise in question this is what will get executed:

```
depcruise --config .dependency-cruiser.cjs src --ignore-known src/app/app.component.ts
```

The `--ignore-known` switch takes an optional argument; the location of the known violations - which it expects to be in json(5) format. This way it'll process the first filename passed behind it to be just that. To prevent dependency-cruiser from processing switches and just processing arguments after add `--`, so the command lint-staged executes would become:

```
depcruise --config .dependency-cruiser.cjs src --ignore-known -- src/app/app.component.ts
```

This works for most cli apps you see it with the `ng-lint-staged` lines in the `.lintstagedrc` as well.

### Always 'cruise' all the things in lint-staged?

With this setup dependency-cruiser will cruise `src` (which already includes src/app/app.component.ts) and not only the staged files. If that's the intention, this will work:

```javascript
// .lintstagedrc.mjs
export default {
  'src/**/*.ts': [
    'ng-lint-staged lint:tour --',
    'depcruise --config .dependency-cruiser.cjs src --ignore-known --',
  ],
  'src/**/*.html': 'ng-lint-staged lint:tour --',
  'projects/my-lib/**/*.ts': [
    'ng-lint-staged lint:lib --',
    'depcruise --config .dependency-cruiser.cjs projects/my-lib --ignore-known --',
  ],
  'projects/my-lib/**/*.html': 'ng-lint-staged lint:lib --',
};
```

> You could also try replacing the `--`s with something that ensures the rest of the line is ignored (a comment mark like `#` springs to mind, or maybe lint-staged has a thing for it), but the extra arguments passed behind `--` won't do harm in any case.

### Only cruise the staged files

What I typically see is that depcruise in lint-staged is run only on the staged files and with a limited depth to boot and not on the rest of the sources. Rationale for that: (1) it's faster (definitely so with larger projects) (2) it won't start emitting errors that have nothing to do with the staged files (your use of --ignore-known cleverly circumvents that - I hadn't thought to use it that way :-) ). Drawback of that approach is that some rules (e.g. no-orphans) will raise false flags - which in itself is circumventable by having a special .lint-staged-dependency-cruiser.cjs which overrides the severity of these rules for example:

```javascript
// .dependency-cruiser-lint-staged.cjs
module.exports = {
    extends: './.dependency-cruiser.cjs',
    forbidden: [
        {
            /**
             * orphan detection i.c.m. max-depth/ let loose on a subset of files
             * is tricky and might generate false positives. As the goal of lint-staged
             * checks is a sanity check (a.o.t. a full blown ci quality check) we
             * lower the severity of the orphans rule so lint-staged won't error
             * out on it.
             */
            name: 'no-orphans',
            severity: 'info' // you could even put 'ignore' in here
        }
    ]
};
```


Thrown together, this will be the thing that will get executed:

```
depcruise --config .dependency-cruiser-lint-staged.cjs --max-depth 1 --ignore-known -- src/app/app.component.ts
```

> the `--max-depth` prevents dependency-cruiser from crawling everything reachable from the 
> modules it gets passed.

or, in .lintstagedrc parlance:

```javascript
// .lintstagedrc.mjs
export default {
  'src/**/*.ts': [
    'ng-lint-staged lint:tour --',
    'depcruise --config .dependency-cruiser-lint-staged.cjs--max-depth 1 --ignore-known --',
  ],
  'src/**/*.html': 'ng-lint-staged lint:tour --',
  'projects/my-lib/**/*.ts': [
    'ng-lint-staged lint:lib --',
    'depcruise --config .dependency-cruiser-lint-staged.cjs --max-depth 1 --ignore-known --',
  ],
  'projects/my-lib/**/*.html': 'ng-lint-staged lint:lib --',
};
```



> Note to self: might be useful to add this to the FAQ or a separate documentation section

**Comment by michahell** [comment]
Hi @sverweij !

Apologies for my late answer, I was sick :(

Thank you so much for your elaborate explanation. I had tried before to add `--` at the end of the command, but this didn't seem to work well in combination with specifying a directory.
I completely forgot to leave out `src` and `projects/my-lib` from the commands! together with `--` this makes total sense.

In my own MWE project, if I do both these things, including the added `max-depth` flag, it works! 🎊 

```
✖ depcruise --config .dependency-cruiser.cjs --max-depth 2 --:

  warn no-orphans: src/app/app.component.spec.ts
  error trigger-depcruiser: src/app/app.component.ts → node_modules/@angular/core/fesm2015/core.mjs

✖ 2 dependency violations (1 errors, 1 warnings). 3 modules, 1 dependencies cruised.
```

However, when I do this in our actual project, I get the following error:
`✖ depcruise --config .dependency-cruiser.cjs --  failed without output (ENOENT).`

I have not been able to figure out what is causing this :( I will try to find some time to narrow this down
I've tried to look for debugging or verbosity flags but I could not find any, does depcruise have those?
